File: app/src/main/java/com/lfork/a98620/lfree/data/goods/local/GoodsLocalDataSource.java
package com.lfork.a98620.lfree.data.goods.local;

import com.lfork.a98620.lfree.data.DataSource;
import com.lfork.a98620.lfree.data.entity.Category;

import org.litepal.crud.DataSupport;

import java.util.List;

/**
 * Created by 98620 on 2018/3/23.
 */

public class GoodsLocalDataSource  {

    private static GoodsLocalDataSource INSTANCE;
//    private ArrayList<ResponseGetUser.UserInfo> mCachedUserList = new ArrayList<>();

    private static final String TAG = "IMRemoteDataSource";

    private GoodsLocalDataSource() {
    }

    public static GoodsLocalDataSource getInstance() {
        if (INSTANCE != null) {
            return INSTANCE;
        } else {
            INSTANCE = new GoodsLocalDataSource();
        }
        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }

    public void getCategories(DataSource.GeneralCallback<List<Category>> callback) {
        List<Category> categories = DataSupport.findAll(Category.class);

        if (categories != null) {
            callback.succeed(categories);
        } else {
            callback.failed("本地数据为空");
        }
    }

}

File: app/src/main/java/com/lfork/a98620/lfree/data/goods/local/GoodsLocalDataSource.kt
package com.lfork.a98620.lfree.data.goods.local

import com.lfork.a98620.lfree.data.DataSource
import com.lfork.a98620.lfree.data.base.entity.Category
import com.lfork.a98620.lfree.data.base.entity.Goods
import com.lfork.a98620.lfree.data.base.entity.GoodsDetailInfo
import com.lfork.a98620.lfree.data.base.entity.Review
import com.lfork.a98620.lfree.data.goods.GoodsDataSource
import org.litepal.crud.DataSupport

/**
 *
 * Created by 98620 on 2018/9/2.
 */
object GoodsLocalDataSource : GoodsDataSource {
    override fun getGoodsList(callback: DataSource.GeneralCallback<List<Goods>>, cursor: String, categoryId: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getUserGoodsList(callback: DataSource.GeneralCallback<List<Goods>>, cursor: String, userId: String) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getGoods(callback: DataSource.GeneralCallback<GoodsDetailInfo>, goodsId: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun deleteGoods(callback: DataSource.GeneralCallback<String>, goodsId: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun uploadGoods(callback: DataSource.GeneralCallback<String>, g: Goods) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun goodsSearch(callback: DataSource.GeneralCallback<List<Goods>>, keyword: String) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun updateGoods(callback: DataSource.GeneralCallback<String>, g: Goods) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun addReview(callback: DataSource.GeneralCallback<Review>, review: Review) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    fun destroyInstance() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getCategories(callback: DataSource.GeneralCallback<List<Category>>) {
        val categories = DataSupport.findAll(Category::class.java)
        if (categories != null) {
            callback.succeed(categories)
        } else {
            callback.failed("本地数据为空")
        }
    }
}
